src/bin/pg_dump/pg_dump.c

```diff
@@ -18547,9 +18567,26 @@ getDependencies(Archive *fout)
 			dobj->objType == DO_TABLE &&
 			refdobj->objType == DO_TYPE)
 			addObjectDependency(refdobj, dobj->dumpId);
+		/*
+		 * T-SQL table-type's template table has implicit dependency upon
```

Comment:
Can it be only applicable to bbf objects? What if there are some pg objects satisfying this condition at the moment or near future?